File: qa-automation/src/pages/DashboardPage.js
const BasePage = require('./BasePage');
const logger = require('../utils/Logger');

class DashboardPage extends BasePage {
    // ----------------------------------------------------------------------
    // Selectors
    // ----------------------------------------------------------------------
    get pageTitle() { return $('.page-title'); }
    get subtitle() { return $('.dashboard-header .text-muted'); }
    get btnStartAnalysis() { return $('button=Start New Analysis'); }
    
    get statCards() { return $$('.stat-card'); }
    get recentScansContainer() { return $('.recent-scans'); }
    get recentScansList() { return $$('.scan-item'); }
    get btnViewAllScans() { return $('.recent-scans .btn-outline'); }
    
    get aiInsightsContainer() { return $('.ai-insights'); }
    get insightItems() { return $$('.insight-item'); }
    
    get noScansMessage() { return $('.scan-list div'); } // When list is empty

    // ----------------------------------------------------------------------
    // Methods
    // ----------------------------------------------------------------------

    /**
     * Navigates directly to dashboard (requires auth session)
     */
    async open() {
        logger.info('Navigating to Dashboard');
        await super.open('dashboard');
        await this.waitForElement(this.pageTitle);
    }

    /**
     * Retrieves the text of the welcome title
     */
    async getWelcomeTitle() {
        await this.waitForElement(this.pageTitle);
        return this.pageTitle.getText();
    }

    /**
     * Retrieves all statistic labels present on the dashboard
     */
    async getStatLabels() {
        await this.waitForElement(this.statCards[0]);
        const labels = await $$('.stat-label');
        return Promise.all(labels.map(async (label) => await label.getText()));
    }

    /**
     * Clicks "Start New Analysis" button
     */
    async clickStartAnalysis() {
        logger.info('Clicking "Start New Analysis" button');
        await this.clickElement(this.btnStartAnalysis);
    }
    
    /**
     * Checks if Recent Scans section is visible
     */
    async isRecentScansVisible() {
        return this.recentScansContainer.isDisplayed();
    }
}

module.exports = new DashboardPage();

File: qa-automation/appium/pages/dashboard_page.py
from .base_page import BaseMobilePage
from selenium.webdriver.common.by import By
import logging

logger = logging.getLogger("AppiumPython")

class MobileDashboardPage(BaseMobilePage):
    # Selectors
    PAGE_TITLE = (By.CSS_SELECTOR, '.page-title')
    SUBTITLE = (By.CSS_SELECTOR, '.dashboard-header .text-muted')
    BTN_START_ANALYSIS = (By.XPATH, '//button[contains(text(), "Start New Analysis") or text()="Start New Analysis"]')
    
    STAT_CARDS = (By.CSS_SELECTOR, '.stat-card')
    RECENT_SCANS_CONTAINER = (By.CSS_SELECTOR, '.recent-scans')
    RECENT_SCANS_LIST = (By.CSS_SELECTOR, '.scan-item')
    
    AI_INSIGHTS_CONTAINER = (By.CSS_SELECTOR, '.ai-insights')
    INSIGHT_ITEMS = (By.CSS_SELECTOR, '.insight-item')

    def get_welcome_title(self):
        """Returns the welcome/header title text"""
        element = self.wait_for_element(*self.PAGE_TITLE)
        return element.text

    def click_start_analysis(self):
        """Taps the Start New Analysis button"""
        logger.info("Appium: Tapping Start New Analysis button")
        self.click_element(*self.BTN_START_ANALYSIS)

    def get_stat_cards_count(self):
        """Returns the count of statistics cards displayed"""
        self.wait_for_element(*self.STAT_CARDS)
        return len(self.driver.find_elements(*self.STAT_CARDS))
